### DISCUSSION TITLE
2-CH RS485 HAT and PI4J

### DISCUSSION THREAD

**Discussion by sergeishugaev** [Pi4J/pi4j#133] [General]
Hello!

I use 2-CH RS485 HAT (https://www.waveshare.com/wiki/2-CH_RS485_HAT) to connect  raspberry pi to indusctrial device via rs485 and Modbus-RTU. The app was writen by Java and work very good. Now I want to read two discrete signals. I try to use PI4J 1.4, but I have a problem. When I write "GpioController gpio = GpioFactory.getInstance();" in my program the serial port which provide by 2-CH_RS485_HAT doesn't work. 
Can I use Pi4J and 2-CH_RS485_HAT driver together? And if it can what will i do for it?
Thanks!

**Comment by sergeishugaev** [comment]
Hello!
Please, help me to solve my problem or give me some advice.
Thanks!

**Comment by FDelporte** [comment]
Sorry, from your question it is not really clear what you already tried in code and what the actual problem is...

Which version of Java or you using? If Java 8, maybe you can use V1.3 and take a look at this example for serial communication:
https://pi4j.com/1.3/example/serial.html

**Comment by sergeishugaev** [comment]
Initially, I needed to exchange information between the raspberry pi and the thermostat via the Modbus-RTU (RS485) protocol. To solve this problem, I bought a 2-CH RS485 HART (two rs485 interfaces on board). com ports of the device are defined on raspberry pi as ttySC0 and ttySC1. I used the jssc library to exchange on the com port, since I already had experience working with this library. Using the library, I open the ttySC0 port and read and write data. I use Java 11.
Later, there was a need to read two discrete signals. I connected the signals to GPIO_0 and GPIO_2 raspberry pi and decided to use the pi4j-v2 library. I implemented the reading of discrete signals, but after that the exchange over RS 485 stopped working on string GpioController gpio = GpioFactory.getInstance(); I understand there is some kind of conflict. And I do not know how to solve it.
Perhaps, you can give some advice?

**Comment by savageautomate** [comment]
```
GpioController gpio = GpioFactory.getInstance(); 
```
This does not look like Pi4J v2 code, but rather Pi4J v.1x.

Pi4J v1.x uses the WiringPi numbering scheme.  So **GPIO_0** is physical/board PIN number 11 on the 40 pin header.

As you can see below, this pin also potentially serves as RTS for serial communication for UART0 and UART1 
_(depending on the ALT mode)_

see: https://pinout.xyz/pinout/pin11_gpio17

<img width="1217" alt="image" src="https://user-images.githubusercontent.com/2251458/140461146-a59c9b58-d336-4f96-b08d-8d1d852bd7ac.png">

So I think there could be a potential conflict when Pi4J is creating the pin and potentially trying to steal the pin from the serial driver.   I would suggest moving off of GPIO_0 if possible and trying another pin.  Maybe GPIO_3.

Thanks, Robert
